### PR TITLE
WIP: add plugin for EDM4hep output from DDG4

### PR DESCRIPTION
**Pull request by gaede**
BEGINRELEASENOTES
- add plugin for EDM4hep output from DDG4
    - writes MCParticles, SimTrackerHits and SimCalorimeterHits
    - similar to LCIO output
- can be chosen in ddsim with "*_edm4hep.root" as outputfile
- requires EDM4hep and PODIO to be available
- only build if `DD4HEP_USE_EDM4hep=TRUE` set
- status: *experimental*

ENDRELEASENOTES

### CODE REVIEW

File: DDG4/python/DDSim/DD4hepSimulation.py [hunk 1]
```diff
@@ -346,6 +346,12 @@ def run(self):
       lcOut.EventParametersString, lcOut.EventParametersInt, lcOut.EventParametersFloat = eventPars
       lcOut.RunNumberOffset = self.meta.runNumberOffset if self.meta.runNumberOffset > 0 else 0
       lcOut.EventNumberOffset = self.meta.eventNumberOffset if self.meta.eventNumberOffset > 0 else 0
+    elif self.outputFile.endswith("edm4hep.root"):
+      e4Out = simple.setupEDM4hepOutput('EDM4hepOutput', self.outputFile)
+#      e4Out.RunHeader = self.meta.addParametersToRunHeader(self)
```
Comment (petricm) [inline DDG4/python/DDSim/DD4hepSimulation.py]: why this comment line?

**Comment by petricm** [comment]
can you please add a test e.g. ddsim gun 1 particle output to edm4hep

File: DDG4/python/DDSim/DD4hepSimulation.py (hunk 1 above)
Comment (gaede) [reply DDG4/python/DDSim/DD4hepSimulation.py]: This is still WIP. There are many lines commented out that deal with event and run headers and meta data parameters as these still don't exist in EDM4hep...

**Comment by vvolkl** [comment]
Hi Frank @gaede,

this looks great already! Maybe it is simpler in terms of dependencies to move this plugin to the EDM4hep repository? Here we'd have to provide a preliminary installation somehow, and it looks like we at least have to add the Run/EventHeaders for a first version.

**Comment by petricm** [comment]
what about adding EDM4hep/master to dev3?

**Comment by petricm** [comment]
Has been agreed that this will be moved to the [EDM4hep Repo](https://github.com/HSF/EDM4hep)

**Comment by vvolkl** [comment]
Supersede by: https://github.com/HSF/EDM4hep/pull/43

**Pull request closed without merging**